juju/juju
# [JUJU-3786] Added controller config domain

This PR introduces the controller config domain, which includes the necessary state API and services to facilitate operations with controller configurations stored in the database. 

## Checklist

~- [ ] [Integration tests](https://github.com/juju/juju/tree/main/tests), with comments saying what you're testing~
~- [ ] [doc.go](https://discourse.charmhub.io/t/readme-in-packages/451) added or updated in changed packages~

## QA steps

```sh
cd domain/controllerconfig/service
go test -check.f serviceSuite

cd ../state
go test -check.f stateSuite

cd ..
go test -check.f controllerconfigSuite
```
